src/service/pubsub.ts

```diff
@@ -16,9 +17,56 @@ export default class PubSubService {
   private constructor() {
     this.initDriver();
     this.initClient();
+    this.startServer();
     this.bind(this);
   }
 
+  public startServer(): void {
+    if (process.env.PUBSUB_HEALTH_SERVER !== 'true') return;
+
+    const port = process.env.PUBSUB_SERVER_PORT || 8080;
+    //create a server object:
+    http
+      .createServer(function (_req, res) {
+        const isHealthy = PubSubService.isHealthy();
+        if (isHealthy) {
+          res.write(`is healthy`); //write a response to the client
+          res.end(); //end the response
+        } else {
+          res.statusCode = 500;
+          res.write(`Not healthy`); //write a response to the client
+          res.end(); //end the response
+        }
+      })
+      .listen(port, () => {
+        Logger.Instance.info(`Pubsub server running on port ${port}`);
+      }); //the server object listens on port 8080
+  }
+
+  public static isHealthy(): boolean {
+    if (PubSubService.status !== 'ready') {
+      Logger.Instance.warn('All subscriptions are not ready yet');
+      return false;
+    }
+
+    const subsState = PubSubService.client.getAllSubscriptionsState();
+    const allSubs = PubSubService.getInstance().getSubscribers();
+
+    const notOpenSubs = subsState
+      .filter((subState) => !subState[1])
+      .map((subState) => subState[0]);
+    const allSubsNotHealthy = !!(
+      notOpenSubs.length && subsState.length === allSubs.length
```

Comment:
Subs are not healthy if either one if them is not open, or open subs count doesn't match total subs

```suggestion
      notOpenSubs.length || subsState.length !== allSubs.length
```